code/game/objects/structures/stool_bed_chair_nest/bed.dm

```diff
@@ -169,14 +169,15 @@
 
 /obj/structure/bed/Destroy()
 	if(current_glue_state == GLUE_STATE_PERMA && is_locking(mob_lock_type))//Don't de-ass someone if it was temporary glue.
-		current_glue_state = GLUE_STATE_NONE
 		var/mob/living/carbon/human/locked = get_locked(mob_lock_type)[1]
 		if(istype(locked) && locked.remove_butt())
 			playsound(src, 'sound/items/poster_ripped.ogg', 100, TRUE)
 			visible_message("<span class='danger'>[locked]'s butt is ripped from their body as \the [src] gets dismantled!</span>")
 			locked.apply_damage(10, BRUTE, LIMB_GROIN)
 			locked.apply_damage(10, BURN, LIMB_GROIN)
 			locked.audible_scream()
+	if(current_glue_state > GLUE_STATE_NONE)
```

Comment:
Can't this check be removed? It should be behave exactly the same if you unconditionally do `current_glue_state = GLUE_STATE_NONE`, no?

Reply:
true.